cmd/migrator/app/migrator.go

```diff
@@ -77,6 +89,65 @@ func Run(ctx context.Context) error {
 		dynamic,
 		migration,
 	)
-	c.Run(ctx)
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	leaderElectionConfig := newLeaderElectionConfig(kubeClient.CoreV1())
+
+	leaderElectionConfig.Callbacks.OnStartedLeading = func(ctx context.Context) {
+		c.Run(ctx)
+	}
+
+	leaderelection.RunOrDie(ctx, *leaderElectionConfig)
+
 	panic("unreachable")
+
+}
+
+func getDefaultResourceLockNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns
+		}
+	}
+	return "default"
+}
+
+func newLeaderElectionConfig(client v1.ConfigMapsGetter) *leaderelection.LeaderElectionConfig {
+
+	id := string(uuid.NewUUID())
+	if hostname, err := os.Hostname(); err == nil {
+		id = hostname + "_" + id
+	}
+
+	if len(*resourceLockNamespace) == 0 {
+	}
+
+	lock := &resourcelock.ConfigMapLock{
+		ConfigMapMeta: metav1.ObjectMeta{
+			Name:      "migrator-lock",
+			Namespace: *resourceLockNamespace,
+		},
+		Client: client,
+		LockConfig: resourcelock.ResourceLockConfig{
+			Identity: id,
+		},
+	}
+
+	return &leaderelection.LeaderElectionConfig{
```

Comment:
be sure that we're gracefully releasing this lock on shutdown.

Reply:
i'm sure we don't

Reply:
the context is not canceled ... i wonder why we have not used the command library we have in library-go